### DISCUSSION TITLE
Find documents where multiple objects has same keys but different values

### DISCUSSION THREAD

**Discussion by RvZEU** [ravendb/ravendb#15823] [Q&A] [answered]
Hi,

i am looking for a solution to query for results where a document has a object, and when two key / values in that object does not pass the query i would like to have that document not in the result.

example i have this query: `From Products as p where p.translations.id = 8 and p.translations.translation_status = 'new'`

example Product document

`{
.....
'translations':[
        {
         "id": 8,
        "translation_status":"translated"
        },
        {
         "id": 6,
        "translation_status":"new"
        },
]
....
}`

In this case it still returns this document as a result, but in this case i only want it to project if the id = 8

Can anybody point me in the right direction.

**Comment by ayende** [comment, marked as answer]
The problem is that you are searching by each property independently.
There are two ways to resolve this, a fanout index or concatenating the field to create a single value during indexing. 
https://ravendb.net/docs/article-page/5.4/csharp/indexes/fanout-indexes

**Comment by RvZEU** [comment]
Thanx! i was just thinking about concatenating the id and status field.
